service/kvmclusterconfig/v1/resource/encryptionkey/delete.go

```diff
@@ -19,5 +21,31 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 // analyses the current and desired state and returns the patch to be applied by
 // Create, Delete, and Update functions.
 func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
+	delete, err := r.newDeleteChangeForDeletePatch(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	patch := framework.NewPatch()
+	patch.SetDeleteChange(delete)
+
+	return patch, nil
+}
+
+func (r *Resource) newDeleteChangeForDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*v1.Secret, error) {
```

Comment:
Is there also newDeleteChangeForUpdatePatch?

Reply:
No. Are you suggesting the name should be `newDeleteChange`?

Reply:
@kopiczko ?

Reply:
We could simplify it that way just like in other resources, yes.

Reply:
I think so, everywhere else it's like that. But it isn't a blocker for me.